src/apps/middleware.js

```diff
@@ -0,0 +1,54 @@
+import { UPDATE_LOCATION } from 'react-router-redux';
+
+import { newBundle, fetchBundleError } from './actions';
+import * as api from '../client-lib/api';
+import RendezvousSocket from '../client-lib/rendezvous-socket';
+
+// this should come from a config options, do this for now
+let outriggerHost = window.location.host;
+
+export const bundleMiddleware = store => {
+    let rendezvous;
+
+    function bundleReceived(bundle) {
+        api.describe(bundle)
+        .then(inputs => {
+            store.dispatch(newBundle(bundle, inputs));
+        })
+    }
+
+    function runModeChanged(runMode) {
```

Comment:
This function might benefit from some extra checks that `runMode` has a valid `path` or `rendezvous` property.